packages/pinia/src/store.ts

```diff
@@ -149,14 +149,10 @@ function createOptionsStore<
       localState,
       actions,
       Object.keys(getters || {}).reduce((computedGetters, name) => {
-        if (__DEV__) {
-          Object.keys(localState || {}).forEach((stateName) => {
-            if (name === stateName) {
-              console.warn(
-                `[🍍]: A getter cannot have the same name as another state property. Rename one of them. Found with "${name}" in store "${id}".`
-              )
-            }
-          })
+        if (__DEV__ && name in (localState || {})) {
```

Comment:
do we really need the `|| {}` check?

```suggestion
        if (__DEV__ && name in localState) {
```